src/useEventListener/useEventListener.stories.ts

```diff
@@ -52,7 +44,7 @@ export const Demo: Story = () => ({
         bind(refs.label, {
```

Comment:
We could also add this complete binding to the hook (if we pass `refs.label` to it), if we feel this will almost always be the same as well.

Reply:
I've moved all of it to the hook, is this what you were aiming for?

Reply:
Yes! :)

Just 2 nits:

1. rename `ref` input param to `logRef`.
2. rename the returned` binding` to `logBinding`.

Reply:
So you are suggesting prefixing all of the exposed functions and variables with the type of hook when returning an object?

So for example on the `useTimeout` hook you would return `startTimeout` and `cancelTimeout` instead of `start` and `stop`?

Reply:
> So you are suggesting prefixing all of the exposed functions and variables with the type of hook when returning an object?

Well, partly. In the examples in the recent PRs, I mostly look at the usage of those returned fields.

Both `state` and `bindings` are things that could be returned from completely different hooks as well, mostly because it's a pattern that is often used inside a component – all components have bindings, and most have some kind of internal state. Chances are high you would use another hook that could also export a `binding` forcing you to "rename" both usages. Especially of such a hook is often used, you probably want to have it (re)named consistently across all usages. So as soon as you see `logBinding`, in the return, you always know where it's from immediately.

However, `start` and `stop` are less likely candidates for other hooks to be exported and used within the same component – unless you invoke multiple timeout hooks, but that would always require renaming. Nor is it a pattern you would see used in all components. So there it might be less needed.
However, I wouldn't mind being a bit more explicit in general :)

> It feels a bit like overkill giving these kind of names to the local functions

If by "local functions" you mean the functions inside the hook, then I could agree. In that case, you probably want to rename them when returning them?

```js
function useFoo() {
  const start = () => ....;
  return { startFoo: start };
}

Reply:
Yeah, that's indeed what I meant by local functions :) 

To be fair though, this leaves some room for interpretation:

> `start` and `stop` are less likely candidates for other hooks

When do we prefix and don't we? Do we all know all of the patterns and when to apply what? 

Maybe we should just be super explicit and always prefix the returned values to ensure no room for interpretation?

Reply:
I would actually suggest returning a tuple, instead of an array; reason being is that you might use the hook multiple times each for a different `ref`, it would also leave the naming completely on the user, if they want to use `log`, `ref` and `binding` as nothing else uses those names in their component, then so be it.

```ts
// multi logs & refs
const [labelLog, labelLogBinding, labelLogs] = useStorybookLog(refs.label);
const [paragraphLog, paragraphLogBinding, paragraphLogs] = useStorybookLog(refs.paragraph);

// single log & ref
const [log, binding, logs] = useStorybookLog(refs.label);
```

Reply:
I like that approach, does @ThaNarie agree? 

It doesn't change the fact that we might want to be super explicit if we do return objects though :)

Reply:
> It doesn't change the fact that we might want to be super explicit if we do return objects though

I don't think that we should have a hard rule for this, and instead treat it on a case by case basis. For example, if we have something like `useTimeout` or `useIntersectionObserver`, that wrap around a browser API. I personally would lean towards naming of whatever is returned (the API) for the hook to match the naming of the browser API.

If we do want at least a rule of thumb, then I'd push towards non-prefixed properties by default.

Reply:
> I would actually suggest returning a tuple, instead of an ~array~ object

This is turning into an interesting conversation :D

Tuples are often used (and excel) when:

* when there are just 2 return values, or at least not too many
* when optional returned values are trailing at the end
* they are super generic (expected to be used multiple times within the same component), so renaming them isn't a chore.
* the order of the returned values are somewhat predictable (e.g. `[value, action]`)

The first two point have similarities with function parameters, using multiple vs an options object.

Returning an object could be useful when:

* You have more than 2-3 returned values
* Most or all can be optional
* The order of the returned values is unpredictable (although TS helps a bit here)

In the given example, the order is reversed from what is commonly done in react:
```js
// [action, state, state]
const [log, binding, logs] = useStorybookLog(refs.label);
// vs usually [state, action]
```

You could change the order in the hook, but what if one of the states becomes optional for some use case, then you'd have to add `_` placeholders.

I don't have a very strong opinion myself, mostly because this specific example sits in the grey middle for me. Personally I tend to lean towards using tuples for more generic and simple hooks, and objects for more specific or complex hooks.

Reply:
Ok, so we need to make a decision here! I would like to get this internal helper hook merged because it's being used in multiple other hooks. 🤓

I think that the tuple approach still makes sense here, if we flip around the return values we match the expected `[value, action]` order and In our scenario, we might even want to remove the `logs` array because we do not have a use case for it at this moment.

Reply:
It's an internal util, so go with your gut on this one :)